### PR TITLE
feat(devops): Release proposal

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
Automation of proposals.

# Changes
- Add a command to create a proposal.

# Tests
This has been used (with minor modification) in the last release.

# TODO
- Builds are not yet reliably reproducible on Mac.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -63,3 +63,20 @@ dfx canister install signer --mode upgrade --upgrade-unchanged --network staging
...
 
 If you are not a controller, you may request a canister upgrade via Orbit. Please contact Leon Tan for the latest Orbit deployment instructions.
+
+## Deploy to Production
+
+- Create a GitHub release with a tag such as `v0.1.2`
+  - Update the GitHub release text. It is recommended to ask the team to review the text.
+  - Ensure that the release has been published.
+- Check out the release commit
+- Delete any old release directory.
```
Comment (lmuntaner) [inline HACKING.md:73]: What are the releases directory? Maybe add an example.

File: HACKING.md [hunk 2]
```diff
@@ -63,3 +63,20 @@ dfx canister install signer --mode upgrade --upgrade-unchanged --network staging
...
+- Install the corresponding `ic-admin`: `./scripts/setup ic-admin`
+- Run: `./scripts/proposal-assets -t $TAG`
+  - Verify that `release/ci` contains the release Wasm and arguments.
+  - Run a docker build locally and verify that the Wasm and argument file hashes match.
```
Comment (lmuntaner) [inline HACKING.md:77]: Add the script on how to run the docker build?

File: HACKING.md [hunk 3]
```diff
@@ -63,3 +63,20 @@ dfx canister install signer --mode upgrade --upgrade-unchanged --network staging
...
+
+## Deploy to Production
+
+- Create a GitHub release with a tag such as `v0.1.2`
+  - Update the GitHub release text. It is recommended to ask the team to review the text.
+  - Ensure that the release has been published.
+- Check out the release commit
+- Delete any old release directory.
+- Install the corresponding `ic-admin`: `./scripts/setup ic-admin`
+- Run: `./scripts/proposal-assets -t $TAG`
```
Comment (lmuntaner) [inline HACKING.md:75]: Is this script present in the repo already?

File: HACKING.md [hunk 4]
```diff
@@ -63,3 +63,20 @@ dfx canister install signer --mode upgrade --upgrade-unchanged --network staging
...
+  - Verify that `release/ci` contains the release Wasm and arguments.
+  - Run a docker build locally and verify that the Wasm and argument file hashes match.
+- Run: `./scripts/proposal-template -t $TAG`
```
Comment (lmuntaner) [inline HACKING.md:78]: Is this script in the repo already?

File: HACKING.md [hunk 5]
```diff
@@ -63,3 +63,20 @@ dfx canister install signer --mode upgrade --upgrade-unchanged --network staging
 ```
 
 If you are not a controller, you may request a canister upgrade via Orbit. Please contact Leon Tan for the latest Orbit deployment instructions.
+
+## Deploy to Production
```
Comment (lmuntaner) [inline HACKING.md:67]: Very clean process! I like it!

**Review by lmuntaner — APPROVED** [review]
I added some comments on the readme.

Not an expert on bash, but LGTM.

File: HACKING.md (hunk 1 above)
Comment (bitdivine) [reply HACKING.md:73]: Done.

File: HACKING.md (hunk 3 above)
Comment (bitdivine) [reply HACKING.md:75]: Yes, it is.

File: HACKING.md (hunk 2 above)
Comment (bitdivine) [reply HACKING.md:77]: Good idea.  Done.

**Pull request merged**